src/highdicom/content.py

```diff
@@ -1506,56 +1834,82 @@ def __init__(
         super().__init__()
 
         # Checks on inputs
-        all_luts = [red_lut, green_lut, blue_lut]
-        for lut in all_luts:
-            if not isinstance(lut, LUT):
+        self._color_luts = {
+            'Red': red_lut,
+            'Green': green_lut,
+            'Blue': blue_lut
+        }
+        for lut in self._color_luts.values():
```

Comment:
I am not sure. However, it don't see why they couldn't have different types. For example, if you encode a continuous colormap for a single color and the other colors remain constant, you may want to provide a segmented LUT for the colors that don't change and provide a LUT for the color that does change (in particular if the change is not linear).

Reply:
The [standard](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.9.html#table_C.7-22) says that, for example RedPaletteColorLUTData is

> Required if segmented data is NOT used in an Image IOD or Color Palette IOD, or if the IOD is a Presentation State IOD.

The use of "segmented data" rather than "red segmented data" to me suggests that all three are the same, but it's not very clear. Also are we sure that we can even use segmented data in a presentation state, given the above?

Reply:
Hmm...fine with me restricting it to either all or none being segmented.

We definitely need support for segmented LUT data in the presentation state. I will not include lots of LUTs with 16-bits entries. If the standard does not support that, then it should be corrected.